### PR TITLE
Add fakerjs

### LINKED ISSUE #1592: Replace manually generated fake data
>Maintaining big arrays of faked data is a bit annoying, especially when the shape changes. We could use something like faker.js to generate them instead. We would probably still need to have functions in common to set up the generation of the data though.

_Originally posted by @petertgiles in https://github.com/GCTC-NTGC/gc-digital-talent/pull/1535_

>Sounds like we should create a story for moving to faker.js data, so we can define and estimate it more concretely, and decide if its worth it.

_Originally posted by @tristan-orourke in https://github.com/GCTC-NTGC/gc-digital-talent/pull/1535#pullrequestreview-839664268_

This issue is created to discuss replacing [our manually generated fake data](https://github.com/GCTC-NTGC/gc-digital-talent/tree/main/common/src/fakeData) with a fake tool like [faker.js](https://github.com/marak/Faker.js/).

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adds a dev-dependency on [faker.js](https://github.com/marak/Faker.js/) to generate fakes and removes much of the manually generated data.

Regarding the questions:

Q) Can it handle enums?
A) Yes, quite nicely.  For example, [here](https://github.com/GCTC-NTGC/gc-digital-talent/blob/206978265f1cddca21404d6555dcbc7d260996ba/common/src/fakeData/fakePoolCandidates.ts#L39).

Q) Is it more effort than just using playground to generate the objects manually one time?
A) I think so, in most cases.  I wrote a script to generate realistic classifications with groupings and that got way too complicated.   For fake data that we have already generated that is very close to reality like CMO assets and classifcations I think it's best to leave it.  But faker can be used to just fake out some fields like IDs and lorems so it doesn't have to be a total conversion.  But for long lists of users or candidates it works very well.

Q) Can it be tree-shaken out?
A)  Yes, I used a bundle analyzer to ensure that it did.  I think my bundle sizes actually went down noticeably because the long lists of candidates and users were removed.

Q) Can it do translations?
A) [It does have localized data](https://github.com/GCTC-NTGC/gc-digital-talent/blob/206978265f1cddca21404d6555dcbc7d260996ba/common/src/fakeData/fakePoolCandidates.ts#L39) but the French corpus seems to be a lot smaller than English.   Hopefully it will grow over time.

Closes #1592

### CODE REVIEW

File: common/src/fakeData/fakeDepartments.ts [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
-      en: "Public Service Commission",
-      fr: "Commission de la fonction publique",
+export default (): Department[] => {
+  faker.seed(0); // repeatable results
+  faker.setLocale("en");
+
+  return [
+    {
+      id: faker.datatype.uuid(),
+      departmentNumber: faker.datatype.number(100),
```
Comment (tristan-orourke) [inline common/src/fakeData/fakeDepartments.ts:11]: Is there a way to ensure each number is unique? Maybe a function that returns an incrementing number?

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Great, this looks very useful. I like being able to set the seed, and I love seeing how it can usefully be combined with manual data.

I'm a tiny bit concerned about generating numbers that should be unique for each item. Collisions would very rarely happen, but at the same time, rare-and-inconsistent issues are annoying. If there's nothing else to be done, I'd increase the range of numbers to more than 100.

File: common/src/fakeData/fakeDepartments.ts (hunk 1 above)
Comment (petertgiles) [reply common/src/fakeData/fakeDepartments.ts]: I found that faker does have a function that guarantees uniqueness.  I applied to the department numbers in my latest commit to demonstrate.  I also removed the upper limit on random number generation.

Strangely, when I started applying it to the uuids it started throwing "failed to generate a unique" value pretty quickly.  It should be practically impossible to generate a collision on uuids so I'm not sure what's happening there.  🤷‍♂️

Comment (tristan-orourke) [reply common/src/fakeData/fakeDepartments.ts]: Huh, that's odd, but yeah we don't need to use it for uuids. The whole point of uuids is that a collision will never happen in practice.

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**